examples/applications/plot_time_series_lagged_features.py

```diff
@@ -0,0 +1,454 @@
+            if key.endswith("05_loss"):
+                mean_pinball_05_loss_list.append(
+                    f"{value.mean():.3f} ± {value.std():.3f}"
+                )
+            if key.endswith("50_loss"):
+                mean_pinball_50_loss_list.append(
+                    f"{value.mean():.3f} ± {value.std():.3f}"
+                )
+            if key.endswith("95_loss"):
+                mean_pinball_95_loss_list.append(
+                    f"{value.mean():.3f} ± {value.std():.3f}"
+                )
+
+    time_list.append(f"{toc - tic:.3f} s")
+
+
+gbrt_mse = HistGradientBoostingRegressor(loss="squared_error")
+evaluate_and_store_results("gbrt_mse", gbrt_mse, X, y, cv=ts_cv)
+
+# %%
+# Model evaluation using `loss="poisson"`
+gbrt_poisson = HistGradientBoostingRegressor(loss="poisson")
+evaluate_and_store_results("gbrt_poisson", gbrt_poisson, X, y, cv=ts_cv)
+
+# %%
+# Modeling Predictive Uncertainty via Quantile Regression
+# -------------------------------------------------------
+# Instead of modeling the expected value of the distribution of
+# `Y|X` like the least squares and Poisson losses do, one could try to
+# estimate quantiles of the conditional distribution.
+#
+# `Y|X=xi` is expected to be a random variable for a given data point `xi`
+# because we expect that the number of rentals cannot be 100% accurately
+# predicted from the features. It can be influenced by other variables
+# not properly captured by the existing lagged features. For instance
+# whether or not it will rain in the next hour cannot be fully be anticipated
+# from the past hours bike rental data. This is what we call aleatoric
+# uncertainty.
+#
+# Quantile regression makes it possible to give a finer description of that
+# distribution without making strong assumptions on its shape.
+#
+# The conditional 5th percentile (a.k.a. 0.05-quantile) can be estimated with:
+gbrt_percentile_05 = HistGradientBoostingRegressor(loss="quantile", quantile=0.05)
+evaluate_and_store_results("gbrt_percentile_05", gbrt_percentile_05, X, y, cv=ts_cv)
+
+# %%
+# The conditional median (0.50-quantile) can be estimated with:
+gbrt_median = HistGradientBoostingRegressor(loss="quantile", quantile=0.5)
+evaluate_and_store_results("gbrt_median", gbrt_median, X, y, cv=ts_cv)
+
+# %%
+# And finally the 0.95 quantile:
+gbrt_percentile_95 = HistGradientBoostingRegressor(loss="quantile", quantile=0.95)
+evaluate_and_store_results("gbrt_percentile_95", gbrt_percentile_95, X, y, cv=ts_cv)
+
+# %%
+# We can now compare the performance of the different models we trained
+dataframe_dict = {
+    "Models": model_names_list,
+    "MAPE": mape_list,
+    "RMSE": rmse_list,
+    "MAE": mae_list,
+    "Mean Pinball 05 Loss": mean_pinball_05_loss_list,
+    "Mean Pinball 50 Loss": mean_pinball_50_loss_list,
+    "Mean Pinball 95 Loss": mean_pinball_95_loss_list,
+    "Time": time_list,
+}
+
+df_model_comparison = pd.DataFrame(dataframe_dict)
+df_model_comparison
+

```

Comment:
Here we can add a paragraph mentioning that the minimal value for the Mean Pinball Loss is indeed achieved by the gbrt with the corresponding quantile. Similarly, the lowest RMSE is obtained by the "gbrt_mse" model, as expected. A bit less intuitive, the best MAPE corresponds to the "gbrt_poisson" model. This is due to the local nature of the Poisson Loss normalization.